lib/containers/common.pm

```diff
@@ -130,4 +159,74 @@ sub test_container_image {
     assert_script_run("grep \"Heartbeat from $image\" '/var/tmp/container_$container'");
 }
 
+# Testing openSUSE based images
+sub test_opensuse_based_image {
+    my $image = $_[0];
+    my ($name, $tag) = split(/:/, $image);
+    $tag = 'latest' if (!defined($tag));
+
+    my $runtime = $_[1] //= "docker";
+
+    my $distri  = $_[2] //= get_required_var("DISTRI");
+    my $version = $_[3] //= get_required_var("VERSION");
+
+    assert_script_run("$runtime image pull $image", timeout => 300);
+    assert_script_run("$runtime image ls | grep '$name' | grep '$tag'");
+    # Running executables works
+    assert_script_run qq{$runtime container run --entrypoint '/bin/bash' --rm $image -c 'echo "I work" | grep "I work"'};
+    # It is the right version
+    if ($distri eq 'sle') {
+        my $pretty_version = $version =~ s/-SP/ SP/r;
+        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server $pretty_version"/ });
+
+        if (is_sle('=12-SP3', $version)) {
+            my $plugin = '/usr/lib/zypp/plugins/services/container-suseconnect';
+            assert_script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin'";
+        } else {
+            my $plugin = '/usr/lib/zypp/plugins/services/container-suseconnect-zypp';
+            assert_script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin -v'";
+            script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lp'", 420;
+            script_run "$runtime container run --entrypoint '/bin/bash' --rm $image -c '$plugin lm'", 420;
+        }
+    } else {
+        validate_script_output qq{$runtime container run --rm $image cat /etc/os-release}, sub { /PRETTY_NAME="openSUSE (Leap )?${version}.*"/ };
+    }
+    # zypper lr
+    assert_script_run("$runtime run --rm $image zypper lr -s", 120);
+    # zypper ref
+    assert_script_run("$runtime run --name refreshed $image sh -c 'zypper -v ref | grep \"All repositories have been refreshed\"'", 120);
+    # Commit the image
+    assert_script_run("$runtime commit refreshed refreshed-image", 120);
+    # Remove it
+    assert_script_run("$runtime rm --force refreshed", 120);
+    # Verify the image works
+    assert_script_run("$runtime run --rm refreshed-image sh -c 'zypper -v ref | grep \"All repositories have been refreshed\"'", 120);
+}
+
+sub perform_container_diff {
+    my $first  = $_[0];
```

Comment:
I did:
```perl
# The container-diff is available only for x86_64 architecture
# The registry.suse.de is not accessible from PC instances
 return if (!check_var('ARCH', 'x86_64') || is_public_cloud());
```
We know about the public cloud issue but it will be more complicated to solve it.
Is this OK for you @Vogtinator?

Reply:
If `registry.suse.de` is not reachable, then both `podman_image` and `docker_image` are broken, as they take the urls from `get_suse_container_urls`. Just not scheduling those modules would be the right option AFAICT

Reply:
That's actually true and those are currently not scheduled so this part of the condition is really not needed.